Source/JavaScriptCore/assembler/ARM64Assembler.h

```diff
@@ -1193,13 +1193,24 @@ class ARM64Assembler {
     ALWAYS_INLINE void ldr_literal(RegisterID rt, int offset = 0)
     {
         CHECK_DATASIZE_ALLOW_CAP();
-        ASSERT(!(offset & 3));
 #if CPU(ARM64_CAPS)
         if (DATASIZE == Datasize_Cap) {
+            // Callers need to ensure that the capability literal is aligned in
+            // memory, but should provide the real `offset` from the load to the
+            // literal, as for any other ldr_literal form.
+            //
+            // We can only actually encode a 16-byte-aligned offset, but the
+            // instruction always aligns (down) the computed address, we can
+            // adjust `offset` here to compensate.
+
+            // Verify that the literal is correctly aligned.
+            ASSERT(((buffer().codeSize() + offset) % sizeof(__intcap_t)) == 0);
+            offset += buffer().codeSize() % sizeof(__intcap_t);
             insn(loadRegisterLiteralCap(encodeLoadRegisterLiteralCapImmediate(offset), rt));
             return;
         }
 #endif
+        ASSERT(!(offset & 3));
```

Comment:
Is this checking for (on Morello) 16 byte alignment? If so, shouldn't it be `ASSERT(!(offset & (sizeof(__intcap_t) - 1)))` or similar?